File: web/src/api/types/collection.ts
export type CollectionRead = {
  id: string;
  name: string;
  description: string | null;
  embedding_model: string;
};

export type CollectionCreate = {
  name: string;
  description?: string;
  embedding_model?: string;
};

export type CollectionUpdate = {
  name?: string;
  description?: string;
  embedding_model?: string;
};

File: rag/src/api/schemas/collection.py
import uuid

from pydantic import BaseModel

from rag.src.common import settings


class CollectionCreate(BaseModel):
    name: str
    description: str | None = None
    embedding_model: str | None = settings.llm.EMBEDDING_MODEL


class CollectionUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    embedding_model: str | None = None


class CollectionRead(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    embedding_model: str

    model_config = {"from_attributes": True}
